src/main/java/org/fundacionjala/coding/VigenereCrypto.java

```diff
@@ -0,0 +1,86 @@
+package org.fundacionjala.coding;
+
+
+/**
+ * Mejorar el cifrado usando el Cifrado de Vigenère:
+ * El cifrado de Vigènere es la secuencia del cifrado de Cesar con transformaciones diferentes.
+ * Usando como clave un texto (String).
+ * Ejemplo:
+ * Supongamos que tenemos la clave: LIMON y el mensaje a cifrar: HOLA FUNDACION JALA
+ * Cada letra de la clave tiene asignado un índice comenzando en 1 hasta 26:
+ * A B C D E F G H I J K L M N O P Q R S T U V W X Y Z
+ * 1 2 3 4 5 6 7 8 9 1 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26
+ * * LIMON = 12, 9, 13, 15, 14
+ * La sustitución funciona de la siguiente forma:
+ * H = Rota 12 (L) = T
+ * O = Rota 9 (I) = X
+ * L = Rota 13 (M) = Y
+ * A = Rota 15 (O) = P
+ * J = Rota 14 (N) = X
+ * A= Rota 12 (L) = M
+ * L= Rota 9 (I) = U
+ * A= Rota 13 (M) = N
+ * S = Rota 15 (O)= H
+ * O = Rota 14 (N)= C
+ * F = Rota 12 (L) = R
+ * T = Rota 9 (I) = C
+ */
+public class VigenereCrypto extends Crypto {
+
+    /**
+     * @param message message to encode
+     * @param keyword key for encode
+     * @return Encrypted String
+     */
+    public String encode(final String message, final String keyword) {
+        StringBuilder msg = new StringBuilder(message);
+        int indice;
+        int cont = 0;
+        for (int i = 0; i < msg.length(); i++) {
+            if (base.contains(msg.charAt(i))) {
+                indice = base.indexOf(msg.charAt(i));
+                indice += base.indexOf(keyword.charAt(cont % keyword.length())) + 1;
+                if (indice > base.size() - 1) {
+                    indice = indice - base.size();
+                } else {
+                    if (indice < 0) {
+                        indice = base.size() + indice;
+                    }
+                }
+                msg.setCharAt(i, base.get(indice));
+                cont++;
+            }
+        }
+        return msg.toString();
+    }
+
+    /**
+     * @param message message to decode vigenere
+     * @param keyword key for decoding
+     * @return decoded String
+     */
+    public String decode(final String message, final String keyword) {
```

Comment:
The encode and decode are pretty similar
only the operation at line 69 is different.
Could you refactor that ?